AC3-2-Final-Retake-Fireblog/AC3-2-Final-Retake-Fireblog/FeedViewController.swift

```diff
@@ -0,0 +1,220 @@
+        
+        storageReference.data(withMaxSize: 1 * 1024 * 1024) { (data, error) in
+            if error != nil {
+                print("ERROR DOWNLOADING IMAGE: \(error!.localizedDescription)")
+            }
+            if let postImage = UIImage(data: data!) {
+                completion(postImage)
+            }
+        }.resume()
+    }
+    
+    
+    // MARK: Lazy Instantiation
+    
+    lazy var tableView: UITableView = {
+        let tv = UITableView()
+        tv.translatesAutoresizingMaskIntoConstraints = false
+        return tv
+    }()
+    
+    // MARK: TableView Data Source Methods
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        let postsSortedByDate = self.posts.sorted() { $0.timestamp > $1.timestamp}
+        return postsSortedByDate.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "FeedCell", for: indexPath) as! FeedTableViewCell
+        let postsSortedByDate = self.posts.sorted() { $0.timestamp > $1.timestamp}
+        
+        let post = postsSortedByDate[indexPath.row]
+        
+        cell.userEmailLabel.text = post.madeByUserWithEmail
+        
+        let timeInterval = TimeInterval(post.timestamp)
+        let dateFromTimeInterval = NSDate(timeIntervalSinceReferenceDate: timeInterval)
+        
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "M/d/yy, h:mm a"
+        
+        let postDateString = dateFormatter.string(from: dateFromTimeInterval as Date)
+        cell.timestampLabel.text = postDateString
+        
+        if post.type == PostType.text.rawValue {
+            cell.postImageView.removeFromSuperview()
+            cell.contentView.addSubview(cell.postTextLabel)
+            
+            let _ = [
+                cell.postTextLabel.widthAnchor.constraint(equalTo: cell.contentView.widthAnchor, multiplier: 0.6),
+                cell.postTextLabel.topAnchor.constraint(equalTo: cell.contentView.topAnchor, constant: 8.0),
+                cell.postTextLabel.trailingAnchor.constraint(equalTo: cell.contentView.trailingAnchor, constant: -8.0),
+                cell.postTextLabel.bottomAnchor.constraint(equalTo: cell.contentView.bottomAnchor, constant: -8.0)
+                ].map{ $0.isActive = true }
+            
+            if let postText = post.text {
+                cell.postTextLabel.text = "\(postText)\n"
+            }
+        }
+        
+        if post.type == PostType.image.rawValue {
+            cell.postTextLabel.removeFromSuperview()
+            cell.contentView.addSubview(cell.postImageView)
+            
+            let _ = [
+                cell.postImageView.widthAnchor.constraint(equalTo: cell.contentView.widthAnchor, multiplier: 0.6),
```

Comment:
Here's part of the reason your images overlapped your text on the iPhone 6. You do make both the image and the text fields proportional to the width but they're not related to one another. It'd be safer to make one proportional to the width and have the other take up the rest of the space. Or you could just eyeball it and make sure you test on the largest and smallest devices.